pageserver/src/page_service.rs

```diff
@@ -574,6 +574,41 @@ enum BatchedFeMessage {
     },
 }
 
+impl BatchedFeMessage {
+    async fn throttle(&mut self, cancel: &CancellationToken) -> Result<(), QueryError> {
+        let (shard, tokens, timers) = match self {
+            BatchedFeMessage::Exists { shard, timer, .. }
+            | BatchedFeMessage::Nblocks { shard, timer, .. }
+            | BatchedFeMessage::DbSize { shard, timer, .. }
+            | BatchedFeMessage::GetSlruSegment { shard, timer, .. } => {
+                (
+                    shard,
+                    // 1 token is probably under-estimating because these
+                    // request handlers typically do several Timeline::get calls.
+                    1,
+                    itertools::Either::Left(std::iter::once(timer)),
+                )
+            }
```

Comment:
Discussed this out of band, but leaving it here too for completeness:

`BatchedFeMessage::Exists` and `BatchedFeMessage::Nblocks` may be served from the relsize cache and not materialize into a get. Should we try to account for this?

Reply:
https://github.com/neondatabase/neon/pull/9962/commits/d679883a085ff411e6c35cfbb5a3a6ca861ef731